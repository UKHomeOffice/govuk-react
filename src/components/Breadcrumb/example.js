import React from 'react';
import Breadcrumb from '.';

export default () => (
  <Breadcrumb>
    <a href="/section">Section 1</a>
  </Breadcrumb>
);
