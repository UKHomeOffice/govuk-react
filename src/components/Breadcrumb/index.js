// https://github.com/alphagov/govuk-frontend/blob/master/src/components/breadcrumb/_breadcrumb.scss
import React from "react";
import glamorous from "glamorous";
import PropTypes from "prop-types";

import { BLACK, GREY_1, YELLOW } from "govuk-colours";
import { MEDIA_QUERIES } from "../../constants/index";

const BreadcrumbContainer = glamorous.div({
  fontFamily: "nta, Arial, sans-serif",
  fontSize: "14px",
  lineHeight: "16px",
  marginTop: "15px",
  marginBottom: "10px",
  WebkitFontSmoothing: "antialiased",
  [MEDIA_QUERIES.LARGESCREEN]: {
    fontSize: "16px",
    lineHeight: "20px"
  }
});

const BreadcrumbList = glamorous.ul({
  margin: 0,
  padding: 0,
  listStyleType: "none",
  display: "block"
});

const BreadcrumbListItem = glamorous.li({
  position: "relative",
  display: "inline-block",
  whiteSpace: "no-wrap",
  marginBottom: "5px",
  marginLeft: "10px",
  paddingLeft: "15px",
  ":first-child": {
    marginLeft: 0,
    paddingLeft: 0,
    "::before": {
      display: "none"
    }
  },
  "::before": {
    content: " ",
    display: "block",
    position: "absolute",
    top: "-1px",
    bottom: "1px",
    left: "-3.31px",
    width: "7px",
    height: "7px",
    margin: "auto 0",
    transform: "rotate(45deg)",
    border: "solid",
    borderWidth: "1px 1px 0 0",
    borderColor: `${GREY_1}`
  },
  "> a": {
    color: `${BLACK}`,
    textDecoration: "underline",
    ":focus": {
      backgroundColor: `${YELLOW}`,
      outline: `3px solid ${YELLOW}`
    }
  }
});

const Breadcrumb = ({ children }) => (
  <BreadcrumbContainer>
    <BreadcrumbList>
      {children.length && children.map ? (
        children.map((child, i) => (
          <BreadcrumbListItem key={child.key || i}>{child}</BreadcrumbListItem>
        ))
      ) : (
        <BreadcrumbListItem>{children}</BreadcrumbListItem>
      )}
    </BreadcrumbList>
  </BreadcrumbContainer>
);

Breadcrumb.propTypes = {
  children: PropTypes.node.isRequired
};

export default Breadcrumb;
