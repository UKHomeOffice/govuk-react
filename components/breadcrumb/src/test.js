import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { createSerializer } from 'jest-emotion';
import * as emotion from 'emotion';

import Breadcrumb from './';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('breadcrumb', () => {
  const emptyNode = [];
  const nullNode = null;
  const wrapper = <Breadcrumb>example</Breadcrumb>;
  const wrapperMultiple = (
    <Breadcrumb>
      <a href="/section">Section 1</a>
      example
    </Breadcrumb>
  );
  const wrapperEmptyNode = (
    <Breadcrumb>
      <a href="/section">Section 1</a>
      {emptyNode}
      {nullNode}
      example
    </Breadcrumb>
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
  });

  it('should render an item in unordered list', () => {
    const output = mount(wrapper);
    expect(output.find('ul')).toHaveLength(1);
  });

  it('should render multiple items in unordered list', () => {
    expect(mount(wrapperMultiple).find('ul li')).toHaveLength(2);
  });

  it('should render an unordered list without ghost/duff children', () => {
    expect(mount(wrapperEmptyNode).find('ul li')).toHaveLength(2);
  });

  it('allows attributes to be set', () => {
    const output = mount(<Breadcrumb id="test" className="test">Crumb</Breadcrumb>);

    expect(output.hasClass('test')).toBe(true);
    expect(output.is('#test')).toBe(true);
  });

  it('matches snapshot', () => {
    expect(mount(wrapperMultiple)).toMatchSnapshot('enzyme.mount');
  });
});
