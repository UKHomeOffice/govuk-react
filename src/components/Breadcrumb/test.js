import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount } from "enzyme";

import Breadcrumb from "./";

describe("breadcrumb", () => {
  const example = "example";
  const wrapper = <Breadcrumb>{example}</Breadcrumb>;

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(wrapper, div);
  });

  it("should render an unordered list", () => {
    const output = shallow(wrapper);
    expect(output.find("ul")).toBeTruthy();
    expect(output.find("ul > li")).toBeTruthy();
  });

  it("matches snapshot", () => {
    expect(mount(wrapper)).toMatchSnapshot(`enzyme.mount`);
  });
});
