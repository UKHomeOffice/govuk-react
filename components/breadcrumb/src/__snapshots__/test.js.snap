// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breadcrumb matches snapshot: enzyme.mount 1`] = `
.emotion-4 {
  margin-bottom: 10px;
}

@media only screen and (min-width:641px) {
  .emotion-4 {
    margin-bottom: 10px;
  }
}

.emotion-3 {
  font-family: "nta",Arial,sans-serif;
  font-size: 14px;
  line-height: 1.1428571429;
  margin-top: 15px;
  -webkit-font-smoothing: antialiased;
  margin-bottom: 10px;
}

@media only screen and (min-width:641px) {
  .emotion-3 {
    font-size: 16px;
    line-height: 1.25;
  }
}

@media only screen and (min-width:641px) {
  .emotion-3 {
    margin-bottom: 10px;
  }
}

.emotion-2 {
  margin: 0;
  padding: 0;
  list-style-type: none;
  display: block;
}

.emotion-0 {
  position: relative;
  display: inline-block;
  white-space: no-wrap;
  margin-bottom: 5px;
  margin-left: 10px;
  padding-left: 15px;
}

.emotion-0:first-child {
  margin-left: 0;
  padding-left: 0;
}

.emotion-0:first-child::before {
  display: none;
}

.emotion-0::before {
  content: '';
  display: block;
  position: absolute;
  top: -1px;
  bottom: 1px;
  left: -3.31px;
  width: 7px;
  height: 7px;
  margin: auto 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  border: solid;
  border-width: 1px 1px 0 0;
  border-color: #6f777b;
}

.emotion-0 > a {
  color: #0b0c0c;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0 > a:focus {
  background-color: #ffbf47;
  outline: 3px solid #ffbf47;
}

<Styled(Breadcrumb)>
  <Breadcrumb
    className="emotion-4"
  >
    <Styled(div)
      className="emotion-4"
    >
      <div
        className="emotion-3"
      >
        <Styled(ul)>
          <ul
            className="emotion-2"
          >
            <Styled(li)
              key="0"
            >
              <li
                className="emotion-0"
              >
                <a
                  href="/section"
                >
                  Section 1
                </a>
              </li>
            </Styled(li)>
            <Styled(li)
              key="1"
            >
              <li
                className="emotion-0"
              >
                example
              </li>
            </Styled(li)>
          </ul>
        </Styled(ul)>
      </div>
    </Styled(div)>
  </Breadcrumb>
</Styled(Breadcrumb)>
`;
